Framework/Source/API/Vulkan/VKState.cpp

```diff
@@ -349,11 +349,10 @@ namespace Falcor
                     // for the given element.
                     uint32_t shaderLocation = pVB->getElementShaderLocation(elemID);
 
-                    auto pAttribReflector = pReflector->getVertexAttributeBySemantic(
-                        pVB->getElementName(elemID));
+                    auto pAttribReflector = pReflector->getVertexAttributeBySemantic(pVB->getElementName(elemID));
                     if (pAttribReflector)
                     {
-                        shaderLocation = (uint32_t) pAttribReflector->location;
+                        shaderLocation = (uint32_t)pAttribReflector->semanticIndex;
```

Comment:
Hmm... I'm not convinced this is right, unless you changed how the Slang reflection data maps over to Falcor.

In the Slang world, if I had HLSL/Slang:

```hlsl
struct Vertex {
    float3 pos : POSITION;
    float3 nrm : NORMAL;
};
```

and then in GLSL I write:

```glsl
in Vertex v;
```

Then `v.nrm` should have a *location* of `1` (same as its "offset" or "binding index" - the Slang API uses these terms interchangeably), but its *semantic index* would be `0` because the semantic `NORMAL` is equivalent to `NORMAL0`.

Maybe the `semanticIndex` field here means something different, but if it is copied over from Slang, I suspect there is a bug here.

Reply:
Bad choice of variable name. I didn't change the logic, just renamed the variable (I forgot what `semanticIndex` means :))